lib/common/huf.h

```diff
@@ -111,6 +111,9 @@ HUF_PUBLIC_API size_t HUF_compress2 (void* dst, size_t dstCapacity, const void*
 #define HUF_WORKSPACE_SIZE_U32 (HUF_WORKSPACE_SIZE / sizeof(U32))
 HUF_PUBLIC_API size_t HUF_compress4X_wksp (void* dst, size_t dstCapacity, const void* src, size_t srcSize, unsigned maxSymbolValue, unsigned tableLog, void* workSpace, size_t wkspSize);
 
+/* The workspace must have alignment at least 4 and be at least this large */
+#define HUF_DECOMPRESS_WORKSPACE_SIZE (2 << 10)
```

Comment:
Ah, I guess it comes from Linux's kbuild robot : 
https://lkml.org/lkml/2017/6/25/65

There, `HUF_readDTableX4()` is estimated at 1636 bytes, which is largest stack usage.
So 2KB seems to be an overestimate.

Are we comfortable with ~400 bytes overestimate ?
We could reduce that, by using `7<<8` instead, which would reduce size by 256 bytes.
To control if it's good enough, one should increase `HUF_TABLELOG_MAX` value, up to its maximum `HUF_TABLELOG_ABSOLUTEMAX`, as some tables will increase size as a result, in order to control if everything still fits.

Of course, the best solution would be to have the actual formula calculating `HUF_DECOMPRESS_WORKSPACE_SIZE` from `HUF_TABLELOG_MAX` (which is the only constant value that may change).